src/constants.ts

```diff
@@ -42,16 +43,27 @@ export const DEFAULT_DELAY = 3000;
 export const DEFAULT_MAX_NUMBER_OF_TRIES = 10;
 export const DEFAULT_CREATE_CDA_TOKEN = true;
 
-export const VERSION_CONTENT_TYPE =
-  INPUT_VERSION_CONTENT_TYPE || DEFAULT_VERSION_CONTENT_TYPE;
-export const FEATURE_PATTERN = INPUT_FEATURE_PATTERN || DEFAULT_FEATURE_PATTERN;
-export const MASTER_PATTERN = INPUT_MASTER_PATTERN || DEFAULT_MASTER_PATTERN;
-export const VERSION_FIELD = INPUT_VERSION_FIELD || DEFAULT_VERSION_FIELD;
 export const DELETE_FEATURE = booleanOr(
   INPUT_DELETE_FEATURE,
   DEFAULT_DELETE_FEATURE
 );
+export const CREATE_CDA_TOKEN = booleanOr(INPUT_CREATE_CDA_TOKEN, DEFAULT_CREATE_CDA_TOKEN);
 export const SET_ALIAS = booleanOr(INPUT_SET_ALIAS, DEFAULT_SET_ALIAS);
+
+// set default actions to respect legacy input ENV's. This will make the new versikon backwards compatible.
+export const DEFAULT_ACTIONS = [
+  "createEnvironment",
+  CREATE_CDA_TOKEN && "createCDAToken",
```

Comment:
from

```node
export const CREATE_CDA_TOKEN = booleanOr(INPUT_CREATE_CDA_TOKEN, DEFAULT_CREATE_CDA_TOKEN);
```

it seems `CREATE_CDA_TOKEN` is a boolean value.

won't this expression be a boolean value after evaluation? do you want to implement a ternary `if` clause here? is that it? or it's some Typescript magic that inserts `createCDAToken` value if `CREATE_CDA_TOKEN` is true?

Reply:
It's some JS/TS magic. If CREATE_CDA_TOKEN is false, then the value with be false. If CREATE_CDA_TOKEN is true, then the value will be the result of the 2nd part of the expression. This combos with the filter below to remove the false entries.